src/test/kotlin/org/rust/ide/actions/macroExpansion/RsShowMacroExpansionActionsTest.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions.macroExpansion
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.project.Project
+import org.intellij.lang.annotations.Language
+import org.rust.lang.RsTestBase
+import org.rust.lang.core.macros.RsExpandedElement
+
+class RsShowMacroExpansionActionsTest : RsTestBase() {
+
+    fun `test no expanison available when macro is not under caret`() = testNoExpansionHappens("""
+        let a = b/*caret*/oo();
+        foo!();
+    """)
+
+    fun `test single level expansion is performed by recursive action`() = testRecursiveExpansion("""
+        macro_rules! foo {
+            () => { println!(); }
+        }
+        fo/*caret*/o!();
```

Comment:
I don't like these carets inside indentifiers. We can place a caret before an identifier - `/*caret*/foo!();`